lib/ansible/modules/cloud/amazon/ec2_vpn_connection.py

```diff
@@ -0,0 +1,199 @@
+
+DOCUMENTATION = '''
+---
+module: ec2_vpn_connection
+short_description: Manage an AWS VPN Connection
+description:
+  - Manage an AWS VPN Connection
+version_added: "2.3"
+author: James Cunningham (@jamescun)
+requirements: [ botocore, boto3 ]
+options:
+  state:
+    description: Create or Delete an AWS VPN Connection.
+    required: False
+    default: present
+    choices: [ 'present', 'absent' ]
+  vpn_connection_id:
+    description: VPN Connection ID (required for state=absent).
+    required: False
+  vpn_gateway_id:
+    description: VPN Gateway ID (required for state=present).
+    required: False
+  customer_gateway_id:
+    description: Customer Gateway ID (required for state=present).
+    require: False
+  static_routes_only:
+    description: Whether the VPN connection uses static routes only. Static routes must be used for devices that don't support BGP.
+    required: False
+    default: False
+'''
+
+EXAMPLES = '''
+# Create a VPN Connection
+- ec2_vpn_connection:
+    state: present
+    vpn_gateway_id: vgw-abcdefgh
+    customer_gateway_id: cgw-abcdefgh
+    region: us-east-1
+  register: vpn
+
+# Delete a VPN Connection
+- ec2_vpn_connection:
+  state: absent
+  vpn_connection_id: vpn-abcdefgh
+'''
+
+RETURN = '''
+vpn.connection:
+  description: Information about the VPN connection
+  returned: success
+  type: complex
+  contains:
+    state:
+      description: The current state of the VPN connection.
+      type: string
+      sample: available
+    type:
+      description: The type of VPN connection.
+      type: string
+      sample: ipsec.1
+    vpn_connection_id:
+      description: The ID of the VPN connection.
+      type: string
+      sample: vpn-abcdefgh
+    vpn_gateway_id:
+      description: The ID of the virtual private gateway at the AWS side of the VPN connection.
+      type: string
+      sample: vgw-abcdefgh
+    customer_gateway_id:
+      description: The ID of the customer gateway at your end of the VPN connection.
+      type: string
+      sample: cgw-abcdefgh
+'''
+
+try:
+    from botocore.exceptions import ClientError
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (boto3_conn, camel_dict_to_snake_dict,
+    ec2_argument_spec, get_aws_connection_info)
+
+
+def create_vpn_connection(module, ec2):
+    filters=[
+        {'Name': 'state', 'Values': ['pending', 'available']},
+        {'Name': 'customer-gateway-id', 'Values': [module.params.get('customer_gateway_id')]},
+        {'Name': 'vpn-gateway-id', 'Values': [module.params.get('vpn_gateway_id')]},
+    ]
+
+    if module.params.get('static_routes_only'):
+        filters.append({'Name': 'option.static-routes-only', 'Values': ['true']})
+
+    vpns = ec2.describe_vpn_connections(Filters=filters)
```

Comment:
Please use a try/except or the AWSRetry decorator here, since this call should provide a nice message to the user about where it failed and what actions they can take.